integrations/tensorflow/bindings/python/pyiree/tf/support/tf_test_utils.py

```diff
@@ -562,18 +567,28 @@ def setUpClass(cls) -> None:
     # Setup the directory for saving compilation artifacts and traces.
     cls._artifacts_dir = _setup_artifacts_dir(cls._module_class.__name__)
 
-  def setUp(self):
-    # Ran before each unit test.
+    # Get the backend information for this test.
+    cls._ref_backend_info = tf_utils.BackendInfo(
+        FLAGS.reference_backend, f"{FLAGS.reference_backend}_ref")
+    cls._tar_backend_infos = get_target_backends()
+
+  def setUp(self) -> None:
+    # Runs before each unit test.
     super().setUp()
-    # Create a CompiledModule for the reference backend and each target backend.
-    ref_backend_info = tf_utils.BackendInfo(FLAGS.reference_backend,
-                                            f"{FLAGS.reference_backend}_ref")
-    self._ref_module = self._compile(ref_backend_info)
-
-    tar_backend_infos = get_target_backends()
-    self._tar_modules = [
-        self._compile(backend_info) for backend_info in tar_backend_infos
-    ]
+    global _global_ref_module
+    global _global_tar_modules
+    if _global_ref_module is None:
```

Comment:
I think we can do this on setUpClass still, just storing it as a global instead of a class variable, no? It would be good to still do per-suite setup in the setUpClass method even if we have to do some shenanigans with storage to workaround this bug.

Reply:
Good point – done!